tests/fixed_point/test_fixed_point_math.py

```diff
@@ -0,0 +1,336 @@
+        assert FixedPointIntegerMath.mul_div_up(int(2.5e27), int(0.5e27), int(1e27)) == 1.25e27
+        assert FixedPointIntegerMath.mul_div_up(int(2.5e18), int(0.5e18), int(1e18)) == 1.25e18
+        assert FixedPointIntegerMath.mul_div_up(int(2.5e8), int(0.5e8), int(1e8)) == 1.25e8
+        assert FixedPointIntegerMath.mul_div_up(369, 271, int(1e2)) == 1000
+        assert FixedPointIntegerMath.mul_div_up(int(1e27), int(1e27), int(2e27)) == 0.5e27
+        assert FixedPointIntegerMath.mul_div_up(int(1e18), int(1e18), int(2e18)) == 0.5e18
+        assert FixedPointIntegerMath.mul_div_up(int(1e8), int(1e8), int(2e8)) == 0.5e8
+        assert FixedPointIntegerMath.mul_div_up(int(2e27), int(3e27), int(3e27)) == 2e27
+        assert FixedPointIntegerMath.mul_div_up(int(2e18), int(3e18), int(3e18)) == 2e18
+        assert FixedPointIntegerMath.mul_div_up(int(2e8), int(3e8), int(3e8)) == 2e8
+        assert FixedPointIntegerMath.mul_div_up(0, int(1e18), int(1e18)) == 0
+        assert FixedPointIntegerMath.mul_div_up(int(1e18), 0, int(1e18)) == 0
+        assert FixedPointIntegerMath.mul_div_up(0, 0, int(1e18)) == 0
+
+    def test_fail_mul_div_up_zero_denominator(self):
+        """Test failure for multiplying two numbers, then dividing by zero"""
+        with self.assertRaises(ValueError):
+            FixedPointIntegerMath.mul_div_up(FixedPointIntegerMath.ONE_18, FixedPointIntegerMath.ONE_18, 0)
+
+    def test_ilog2(self):
+        """Test integer log base 2"""
+        # pylint: disable=protected-access
+        assert FixedPointIntegerMath.ilog2(0) == 0
+        assert FixedPointIntegerMath.ilog2(1) == 0
+        assert FixedPointIntegerMath.ilog2(2) == 1
+        assert FixedPointIntegerMath.ilog2(3) == 1
+        assert FixedPointIntegerMath.ilog2(4) == 2
+        assert FixedPointIntegerMath.ilog2(8) == 3
+        assert FixedPointIntegerMath.ilog2(16) == 4
+        assert FixedPointIntegerMath.ilog2(32) == 5
+        assert FixedPointIntegerMath.ilog2(64) == 6
+        assert FixedPointIntegerMath.ilog2(128) == 7
+        assert FixedPointIntegerMath.ilog2(256) == 8
+        assert FixedPointIntegerMath.ilog2(512) == 9
+        assert FixedPointIntegerMath.ilog2(1024) == 10
+        assert FixedPointIntegerMath.ilog2(2048) == 11
+        assert FixedPointIntegerMath.ilog2(4096) == 12
+        assert FixedPointIntegerMath.ilog2(8192) == 13
+        assert FixedPointIntegerMath.ilog2(16384) == 14
+        assert FixedPointIntegerMath.ilog2(32768) == 15
+        assert FixedPointIntegerMath.ilog2(65536) == 16
+        assert FixedPointIntegerMath.ilog2(131072) == 17
+        assert FixedPointIntegerMath.ilog2(262144) == 18
+        assert FixedPointIntegerMath.ilog2(524288) == 19
+        assert FixedPointIntegerMath.ilog2(1048576) == 20
+
+    def test_ln(self):
+        """Test integer natural log"""
+        tolerance = 1e-15  # TODO: Should allow for error up to 1e-18
```

Comment:
is this sticking around for now?

Reply:
Yes, it's on the list of things to fix https://github.com/delvtech/elf-simulations/issues/112